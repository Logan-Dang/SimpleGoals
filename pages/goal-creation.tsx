import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
  TextInput,
} from 'react-native';

export default function NewGoal() {
  return <View style={styles.container}></View>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    height: '100%',
    margin: 0,
  },

  title: {
    fontSize: 24,
  },
});
