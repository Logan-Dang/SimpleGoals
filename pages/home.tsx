import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
} from 'react-native';
export default function Home() {
  let currentTitle;
  let defaultSub;
  let backgroundImage = require('../assets/sunrise.png');
  let screenWidth = Dimensions.get('window').width;
  if (getTime() >= 6 && getTime() < 12) {
    currentTitle = 'Good Morning!';
    defaultSub = "Let's get started";
  } else if (getTime() >= 12 && getTime() <= 18) {
    currentTitle = 'Good Afternoon!';
    defaultSub = 'How was lunch?';
  } else {
    currentTitle = 'Good Evening!';
    defaultSub = "Let's review your day";
  }
  return (
    <View style={styles.container}>
      <ImageBackground
        source={backgroundImage}
        resizeMode='cover'
        style={styles.background}>
        <Text style={styles.title}>{currentTitle}</Text>
        <Text style={styles.subtitle}>{defaultSub}</Text>
        <Text>Goal 1</Text>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    height: '100%',
    margin: 0,
  },

  title: {
    fontSize: 24,
  },

  subtitle: {
    fontSize: 16,
    marginTop: '3%',
  },

  background: {
    alignItems: 'center',
    width: Dimensions.get('screen').width,
    height: Dimensions.get('screen').height,
    paddingTop: '15%',
    margin: 0,
  },
});

function getTime() {
  const currentTime = new Date();
  const time = currentTime.getHours();
  return time;
}
