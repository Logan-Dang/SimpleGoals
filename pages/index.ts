export { default as Home } from './home';
export { default as NewGoal } from './new-goal';
