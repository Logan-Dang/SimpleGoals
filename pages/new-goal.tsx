import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Dimensions,
  TextInput,
  KeyboardAvoidingView,
  findNodeHandle,
} from 'react-native';

import { Button, SomeText } from '../components';

import { FieldError } from 'react-hook-form';

export default function NewGoal() {
  return (
    <View>
      <TextInput style={styles.textStyle} value={'please work'} />
    </View>
  );
}

const styles = StyleSheet.create({
  textStyle: {
    marginTop: 200,
  },
});
