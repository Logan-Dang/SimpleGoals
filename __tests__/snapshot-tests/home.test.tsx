import { create } from 'react-test-renderer';
import { Home } from '../../pages';

it('renders correctly', () => {
  jest.mock('expo-sqlite');
  const tree = create(
    <Home route={undefined} navigation={undefined} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  jest.restoreAllMocks();
});
