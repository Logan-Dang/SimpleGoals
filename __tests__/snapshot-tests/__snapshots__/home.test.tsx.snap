// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "height": "100%",
      "margin": 0,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "height": 1334,
        "margin": 0,
        "paddingTop": "15%",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Good Afternoon!
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginTop": "3%",
        }
      }
    >
      How was lunch?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "JustifyContent": "center",
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "elevation": 3,
          "paddingHorizontal": 32,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        New Goal
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "JustifyContent": "center",
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 4,
          "elevation": 3,
          "paddingHorizontal": 32,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          }
        }
      >
        Log Goals
      </Text>
    </View>
  </View>
</View>
`;
